contract/models/contract.py

```diff
@@ -416,7 +416,7 @@ def _get_lines_to_invoice(self, date_ref):
             lambda l: not l.is_canceled
             and l.recurring_next_date
             and l.recurring_next_date <= date_ref
-        )
+        ).sorted('sequence')
```

Comment:
Why this?

Reply:
I do this to guarantee that, when generating the invoice, each product line is under the corresponding section line and so what @chienandalu  commented in his review doesn't happen

Reply:
But you should take the order of the model, not this one (`env['contract.line']._order`), and are you sure order is not preserved on filtering?

Reply:
I was really wrong, the solution to the problem has nothing to do with reordering recordsets, haha. The 'sequence' field of the contract line was not being passed to create the invoice line. Thank you @pedrobaeza